daemon/algod/api/server/router.go

```diff
@@ -74,18 +75,26 @@ func registerHandlers(router *echo.Echo, prefix string, routes lib.Routes, ctx l
 
 // NewRouter builds and returns a new router with our REST handlers registered.
 func NewRouter(logger logging.Logger, node APINodeInterface, shutdown <-chan struct{}, apiToken string, adminAPIToken string, listener net.Listener, numConnectionsLimit uint64) *echo.Echo {
-	if err := tokens.ValidateAPIToken(apiToken); err != nil {
-		logger.Errorf("Invalid apiToken was passed to NewRouter ('%s'): %v", apiToken, err)
-	}
+	// check admin token and init admin middleware
 	if err := tokens.ValidateAPIToken(adminAPIToken); err != nil {
 		logger.Errorf("Invalid adminAPIToken was passed to NewRouter ('%s'): %v", adminAPIToken, err)
 	}
 	adminMiddleware := []echo.MiddlewareFunc{
 		middlewares.MakeAuth(TokenHeader, []string{adminAPIToken}),
 	}
+
+	// check public api tokens and init public middleware
 	publicMiddleware := []echo.MiddlewareFunc{
 		middleware.BodyLimit(MaxRequestBodyBytes),
-		middlewares.MakeAuth(TokenHeader, []string{adminAPIToken, apiToken}),
+	}
+	if apiToken == "" {
+		logger.Warn("Running with empty apiToken")
+	} else {
+		if err := tokens.ValidateAPIToken(apiToken); err != nil {
+			logger.Errorf("Invalid apiToken was passed to NewRouter ('%s'): %v", apiToken, err)
```

Comment:
Dumb question: if the token is invalid, why _should_ we continue with initialization? If we think we should, why still pass in the invalid token as something to be matched against?